arroyo-console/src/routes/connections/ChooseConnector.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+  Box,
+  Breadcrumb,
+  BreadcrumbItem,
+  BreadcrumbLink,
+  Button,
+  Card,
+  CardBody,
+  CardHeader,
+  Container,
+  Heading,
+  Image,
+  Link,
+  SimpleGrid,
+  Spacer,
+  Stack,
+  Text,
+} from '@chakra-ui/react';
+import { ApiClient } from '../../main';
+import { useNavigate } from 'react-router-dom';
+import { useConnectors } from '../../lib/data_fetching';
+
+export function ChooseConnector({ client }: { client: ApiClient }) {
+  const navigate = useNavigate();
+  const { connectors } = useConnectors(client);
+
+  return (
```

Comment:
I think a good pattern to follow is to define several smaller pieces separately and use the return statement just to put them together at a high level. It makes it much more readable and easier to deal with. [CreatePipeline.tsx](https://github.com/ArroyoSystems/arroyo/blob/master/arroyo-console/src/routes/pipelines/CreatePipeline.tsx) is a good example of that.